src/crdt.rs

```diff
@@ -611,6 +622,10 @@ impl Crdt {
                     //come up with a cleaner solution for this when sender signatures are checked
                     outblob.set_id(from.id).expect("blob set_id");
                 }
+
+                drop(rblob);
```

Comment:
Instead of dropping that, can you pass it back to the recycler?

Reply:
No problem, I'll add some tests. Here, I just wanted to release the read lock so that I can obtain the write lock to modify the blob, not free the underlying blob to the recycler.